core/lib/tasks/migrations/create_ledger_entries_for_store_credits.rake

```diff
@@ -0,0 +1,20 @@
+namespace :solidus do
+  namespace :migrations do
+    namespace :create_ledger_entries_for_store_credits do
+      task up: :environment do
+        print 'Creating opening ledger entries for store credits...'
+        Spree::StoreCredit.find_each do |store_credit|
+          current_balance_amount = store_credit.amount_remaining + store_credit.amount_authorized
+          store_credit.store_credit_ledger_entries.create!({
+            amount: current_balance_amount
+          })
```

Comment:
I would still prefer that this didn't use the standard AR models as that will cause problems if the implementation of `amount_remaining` ever changes. It seems really likely that that method will chnage, given that it might be better implemented in terms of summing these ledger entries.

Instead I'd like to see raw SQL which shouldn't cause any issues in the future.

``` ruby
execute <<-SQL
INSERT INTO spree_store_credit_ledger_entries
  (store_credit_id, amount, created_at, updated_at)
  SELECT id, amount - amount_used, '#{now}', '#{now}'
  FROM spree_store_credits
  WHERE invalidated_at IS NULL
SQL
execute <<-SQL
INSERT INTO spree_store_credit_ledger_entries
  (store_credit_id, amount, created_at, updated_at)
  SELECT id, 0, '#{now}', '#{now}'
  FROM spree_store_credits
  WHERE invalidated_at IS NOT NULL
SQL
```